includes/ziti/model_collections.h

```diff
@@ -80,8 +78,37 @@ for (var(e,line) = model_map_iterator(map), k = model_map_it_key(var(e,line)), v
 
 #define MODEL_MAP_FOR(it, m) for(model_map_iter it = model_map_iterator(&(m)); (it) != NULL; (it) = model_map_it_next(it))
 
+typedef struct model_list_s {
+    struct model_list_impl_s *impl;
+} model_list;
+
+typedef void *model_list_iter;
+
+ZITI_FUNC size_t model_list_size(const model_list *l);
+
+ZITI_FUNC void model_list_append(model_list *l, void *el);
```

Comment:
i expect we're only adding the funcs we need but... i just wonder if there any way to append another model_list, or to grow the list by a factor instead of one at a time

Reply:
I am not sure what you mean -- grow the list without adding elements?

Reply:
just an optimization... if i knew i was appending 10 elements or 100, pre calloc the list... some impls use an algorithm like "grow the list 40% bigger each time you run out of space"... nbd

Reply:
you're making assumption(wrong) about the list implementation. It is not an array list